pkg/gadgets/socket-collector/tracer/tracer.go

```diff
@@ -183,15 +173,18 @@ func RunCollector(pid uint32, filter *gadgetv1alpha1.ContainerFilter) (string, e
 				return "", err
 			}
 
-			sockets = append(sockets, Socket{
-				Protocol:            proto,
-				LocalAddress:        parseIPv4(src),
-				LocalPort:           srcp,
-				RemoteAddress:       parseIPv4(dest),
-				RemotePort:          destp,
-				Status:              status,
-				KubernetesNamespace: filter.Namespace,
-				KubernetesPod:       filter.Podname,
+			sockets = append(sockets, socketcollectortypes.Event{
+				Event: eventtypes.Event{
+					Node:      node,
```

Comment:
Does this work on multi-node cluster? I wonder if inconsistent custom resource would return the wrong result:
```
apiVersion: gadget.kinvolk.io/v1alpha1
kind: Trace
metadata:
  name: socket-collector
  namespace: gadget
spec:
  node: node1
  gadget: socket-collector
  filter:
    namespace: default
    podname: mypod
    node: node2
```

Reply:
It is not possible to deploy a trace with `filter.node` because, from a quick analysis, we don't support it in `ContainerFilter`. While the `Spec.Node` is guarantee to be the same node where the `controller` is running because of this check, is it correct? Sorry, I am still studying this part of the code.

https://github.com/kinvolk/inspektor-gadget/blob/e6449aa674a8e9b68ab1a759c0f6ad2a9c349065/pkg/controllers/trace_controller.go#L82-L85

On the other hand, using the `kubectl-gadget` is not possible to create such a inconsistency (At least from what I know :stuck_out_tongue_closed_eyes:) so the current code should guarantee that `trace.Spec.Node` always contains the correct information in it (Coherent with `--node` in case it was specified).

However, given that I don't have a multi-node cluster (In my TODO list), I tried to filter by a non-existent node and the error is not being handled:
```
./kubectl-gadget-linux-amd64 socket-collector -n default -p my-pod --node non-existent-node
FATA[0000] Error getting traces from all nodes           args="[]" command="kubectl-gadget socket-collector"
```
Maybe we could be done in this loop, correct?
https://github.com/kinvolk/inspektor-gadget/blob/e6449aa674a8e9b68ab1a759c0f6ad2a9c349065/cmd/kubectl-gadget/utils/trace.go#L139

Reply:
There are 2 bugs:
* The loop you mentioned does not check when the node passed as filter is not in the list of node
* The error message is not great: we have the same error when the timeout happens before the gadget pod writes the result in Trace.Status.

Could you open issues for thoses?

Reply:
Issues:

- https://github.com/kinvolk/inspektor-gadget/issues/251
- https://github.com/kinvolk/inspektor-gadget/issues/252